redpanda/cloud/ratelimiter.go

```diff
@@ -0,0 +1,94 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	limitPeriod = 60.0
+	burstPeriod = 10.0
+)
+
+type rateLimiter struct {
+	limiter *rate.Limiter
+}
+
+func newRateLimiter(limit int) *rateLimiter {
+	return &rateLimiter{
+		limiter: rate.NewLimiter(rate.Limit(limit/limitPeriod), limit/burstPeriod),
+	}
+}
+
+func parseRateLimit(header string) (limit, remaining int, reset time.Duration, err error) {
```

Comment:
Please document what it does and if possible add a test for this function. Adding the expected format of the header to parse can help here

Reply:
Test for the function is part of the PR? https://github.com/redpanda-data/terraform-provider-redpanda/blob/5500475701c29e28b7053ca8b6633eec82e05794/redpanda/cloud/ratelimiter_test.go

Reply:
The documentation helps for future maintainability and to understand what are we trying to parse.

btw, the test you linked is a test for the ratelimiter (which uses the `parseRateLimit`). However, it does test the underlying function